java/src/main/java/ai/onnxruntime/OrtUtil.java

```diff
@@ -383,15 +383,48 @@ private static void flattenString(Object[] input, ArrayList<String> output) {
   }
 
   /**
-   * Stores a boxed primitive in a single element array of the boxed type. Otherwise returns the
-   * input.
+   * Stores a boxed primitive in a single element array of the unboxed type.
    *
+   * <p>If it's not a boxed primitive then it returns null.
+   *
+   * @param javaType The type of the boxed primitive.
    * @param data The boxed primitive.
-   * @return The boxed primitive in an array.
+   * @return The primitive in an array.
    */
-  static Object convertBoxedPrimitiveToArray(Object data) {
-    Object array = Array.newInstance(data.getClass(), 1);
-    Array.set(array, 0, data);
-    return array;
+  static Object convertBoxedPrimitiveToArray(OnnxJavaType javaType, Object data) {
+    switch (javaType) {
+      case FLOAT:
```

Comment:
>case FLOAT: [](http://example.com/codeflow?start=6&length=11)

Still no support for FLOAT16 and BFLOAT16? We introduced it in C#.

Reply:
Not as part of this PR. C# has value types so there is a sensible thing to represent it. In Java we could use a `short` and it'll be unpleasant to use, or we can use some reference type and it'll be slower.

I'm considering adding a way to pass in a `float` or `FloatBuffer` and have the native code downcast that to fp16 or bfloat16 as an input. That would be an alternative that wouldn't necessarily be any slower, but it would be ugly if users are taking the output of one model that's fp16 and passing it into another.

Basically there's no really good way of doing this in Java yet, all the options are bad in one way or another. We can pick a way to support and then note that it might evolve as the Java platform changes, or just not support it until the platform has better functionality.

Reply:
Here is my suggestion. Pass it as primitive short type array via separate entry point(s) that would telll ORT C API that this, in fact, is a float16/bfloat16.

Reply:
Sure. I'd prefer to do that in a separate PR though. This one is quite narrowly focused on fixing the bug, whereas adding support for fp16 is going to hit a lot of different places.